qmlti/lib.php

```diff
@@ -428,16 +506,105 @@ function get_assessment_list_by_administrator($id) {
 
   }
 
+/* 
+ * Boolean check for coaching report availability
+ *
+ * returns TRUE if coaching report is valid
+ */
+function is_coaching_report_available($db, $resource_link_id, $assessment_id, $participant_name) {
+  $data_connector = LTI_Data_Connector::getDataConnector(TABLE_PREFIX, $db, DATA_CONNECTOR);
+   if ($data_connector->ReportConfig_loadAccessible($resource_link_id, $assessment_id) == 1) {
+      $result_id = get_result_id($participant_name);
+      if (( $result_id != FALSE ) && (count( (array)$result_id) != 0 )) {
```

Comment:
do you really need to check both of these? if so why not put this logic in PerceptionSoap to make it easier to use instead of needing to duplicate this logic each time